Adding description of data collected by telemetry (opencti:issue/6292) (#161)
OpenCTI-Platform/docs: docs/deployment/overview.md

!!! note "Clustering"
    
    To have more details about deploying OpenCTI and its dependencies in cluster mode, please read the [dedicated section](clustering.md).

## Data collection

The Application collects statistical data related to its usage, including but not limited to:

> - The edition of OpenCTI that is used: either Community or Enterprise edition

Reviewer: ```suggestion
- If Enterprise Edition is enabled, and the Enterprise Edition activation date if applicable
```

> - Number of active users (periodically count)

Reviewer: ```suggestion
- The number of active users (periodically count)
```

> - Number of nodes in the platform

Reviewer: ```suggestion
- The number of platform instances
```

### Purpose of Data Collection:

The collected data is used for the following purposes:

> - Improving the functionality and performance of the Application.

Reviewer: ```suggestion
- Improving the functionality and performance of the application.
```

- Analyzing user behavior to enhance user experience.
- Generating aggregated and anonymized statistics for internal and external reporting.

### Data Anonymization:

> Personally identifiable information is not collected by the Application. All data collected is anonymized and aggregated to protect the privacy of individual users.

Reviewer: ```suggestion
Personally identifiable information is not collected by the application. All the collected is anonymized and aggregated to protect the privacy of individual users.
```
